solvebio/query.py

```diff
@@ -914,7 +918,10 @@ def join(self, query_b, key, key_b=None, prefix="b_", always_prefix=False):
             explode_fields.append(name)
 
             if name in existing_field_names:
-                logger.warning("Field '{}' will be overwritten in the join results".format(name))
```

Comment:
How about this, instead of just showing a warning, lets raise an exception that says `Field '{}' found in both queries, please use a different prefix` then the user is just forced to set a new prefix and the query wont crash.